### PR TITLE
Use Hatchling build-backend and fix README for PyPI

### LINKED ISSUE #995: Automatic module discovery in pyproject.toml
No description provided.

### LINKED ISSUE #1048: Broken images on PyPI
Maybe they need absolute links?
image

### PR DESCRIPTION
**Pull request by maresb**
## Description

See individual commits for a breakdown.

* Switch from setuptools to [hatchling](https://pypi.org/project/hatchling/) because setuptools is really crusty. (Note that Hatchling is just a build backend so it doesn't necessitate the use of Hatch or any other build front-end; you can keep using `pip` and `python -m build` as normal.)
* Clean up a bunch of the README links
* Set up some regex to automatically convert links

The result of the regex can be previewed with `pipx run hatch-fancy-pypi-readme` and the diff is

```diff
3c3
< ![PyMC-Marketing Logo](docs/source/_static/marketing-logo-light.jpg)
---
> ![PyMC-Marketing Logo](https://raw.githubusercontent.com/pymc-labs/pymc-marketing/main/docs/source/_static/marketing-logo-light.jpg)
30c30
<     <img src="docs/source/_static/labs-logo-light.png" width="50%" />
---
>     <img src="https://raw.githubusercontent.com/pymc-labs/pymc-marketing/main/docs/source/_static/labs-logo-light.png" width="50%" />
33c33
< For businesses looking to integrate PyMC-Marketing into their operational framework, [PyMC Labs](https://www.pymc-labs.com) offers expert consulting and training. Our team is proficient in state-of-the-art Bayesian modeling techniques, with a focus on Marketing Mix Models (MMMs) and Customer Lifetime Value (CLV). For more information see [here](README.md#-schedule-a-free-consultation-for-mmm--clv-strategy).
---
> For businesses looking to integrate PyMC-Marketing into their operational framework, [PyMC Labs](https://www.pymc-labs.com) offers expert consulting and training. Our team is proficient in state-of-the-art Bayesian modeling techniques, with a focus on Marketing Mix Models (MMMs) and Customer Lifetime Value (CLV). For more information see [here](https://github.com/pymc-labs/pymc-marketing/tree/main/README.md#-schedule-a-free-consultation-for-mmm--clv-strategy).
57c57
< We provide a `Dockerfile` to build a Docker image for PyMC-Marketing so that is accessible from a Jupyter Notebook. See [here](scripts/docker/README.md) for more details.
---
> We provide a `Dockerfile` to build a Docker image for PyMC-Marketing so that is accessible from a Jupyter Notebook. See [here](https://github.com/pymc-labs/pymc-marketing/tree/main/scripts/docker/README.md) for more details.
114c114
< ![](docs/source/_static/mmm_plot_components_contributions.png)
---
> ![](https://raw.githubusercontent.com/pymc-labs/pymc-marketing/main/docs/source/_static/mmm_plot_components_contributions.png)
119c119
<     <img src="docs/source/_static/mmm_plot_plot_channel_contributions_grid.png" width="80%" />
---
>     <img src="https://raw.githubusercontent.com/pymc-labs/pymc-marketing/main/docs/source/_static/mmm_plot_plot_channel_contributions_grid.png" width="80%" />
169c169
< ![](docs/source/_static/expected_purchases.png)
---
> ![](https://raw.githubusercontent.com/pymc-labs/pymc-marketing/main/docs/source/_static/expected_purchases.png)
175c175
< PyMC-Marketing is and will always be free for commercial use, licensed under [Apache 2.0](LICENSE). Developed by core developers behind the popular PyMC package and marketing experts, it provides state-of-the-art measurements and analytics for marketing teams.
---
> PyMC-Marketing is and will always be free for commercial use, licensed under [Apache 2.0](https://github.com/pymc-labs/pymc-marketing/tree/main/LICENSE). Developed by core developers behind the popular PyMC package and marketing experts, it provides state-of-the-art measurements and analytics for marketing teams.
199a200
> 
```

## Related Issue

- [x] Closes #1048
- [x] Closes #995

## Checklist

- [x] Checked that [the pre-commit linting/style checks pass](https://docs.pymc.io/en/latest/contributing/python_style.html)
- [ ] Included tests that prove the fix is effective or that the new feature works
- [ ] Added necessary documentation (docstrings and/or example notebooks)
- [x] If you are a pro: each commit corresponds to a [relevant logical change](https://wiki.openstack.org/wiki/GitCommitMessages#Structural_split_of_changes)

## Modules affected

- [ ] MMM
- [ ] CLV

## Type of change

- [ ] New feature / enhancement
- [ ] Bug fix
- [ ] Documentation
- [x] Maintenance
- [ ] Other (please specify):

----
📚 Documentation preview 📚: https://pymc-marketing--1051.org.readthedocs.build/en/1051/

### CODE REVIEW

**Comment by wd60622** [comment]
Not very familiar but logically makes sense

File: pyproject.toml [hunk 1]
```diff
@@ -80,7 +79,12 @@ test = [
     "mlflow>=2.0.0",
 ]
 
-[tool.setuptools]
+[tool.hatch.build.targets.sdist]
+exclude = [
+    "docs",
+]
+
+[tool.hatch.build.targets.wheel]
```
[the remark this answers is not in the document]
Comment (maresb) [reply pyproject.toml]: By default, good question! :joy: 

e62283c

**Comment by wd60622** [comment]
Does pipx run hatch-fancy-pypi-readme have to be manually run again?

**Comment by maresb** [comment]
Nope, the only thing that command does is to print a preview of the modified README to stdout.

That command is useful for checking if the PyPI version of the README will come out as expected.

**Comment by juanitorduz** [comment]
Thanks @maresb ! Out of curiosity ... is this part of a plan to move to `pixi` or in your opinion is not necessary (I genuinely do not know)

**Comment by maresb** [comment]
It's independent. The pixi stuff should not and will not touch `pyproject.toml`. The environment management (pixi config) is independent from the package definition (pyproject.toml) except for the environment including the package.

Pixi does allow its config to be read from `pyproject.toml` by prepending `tool.pixi.` to all the table names, but I'm very opinionated that this is an awful idea due to the potential for confusion.

**Comment by maresb** [comment]
Ah I meant to link this. Here's what adding pixi looks like: https://github.com/pymc-labs/CausalPy/pull/412

**Comment by juanitorduz** [comment]
I think this PR is failing because an unrelated issue (mmm.save("model.nc") in the example notebook). I had commented that line before so I'll patch this again tonight)

**Comment by maresb** [comment]
Thanks @juanitorduz and @wd60622! All green now.

**Comment by juanitorduz** [comment]
Amazing! Thanks @maresb !

**Pull request merged**